aten/src/ATen/templates/RegisterBackendSelect.cpp

```diff
@@ -29,13 +30,13 @@ bool is_pinned(const Tensor& self, std::optional<at::Device> device) {
     return false;
   }
   // TODO: fetch scalar type from Tensor? But it doesn't really matter...
-  DispatchKeySet _dk = c10::DispatchKeySet(c10::computeDispatchKey(c10::nullopt, self.layout(), device.value_or(at::kCUDA)));
+  DispatchKeySet _dk = c10::DispatchKeySet(c10::computeDispatchKey(c10::nullopt, self.layout(), device.value_or(at::getAccelerator(false).value_or(at::kCUDA))));
```

Comment:
The `is_pinned` method is used anywhere even for a cpu-only PyTorch. So, here we fall back to CUDA (`CUDAHooks.isPinnedPtr()` returns `false` when cpu-only is built) if there is no available accelerator.